DOCUMENTATION.rst

```diff
@@ -957,20 +958,20 @@ A possible parser and callable behavior would be:
     >>> value = {
     ...     "class_path": "SGD",
     ...     "init_args": {
-    ...         "lr": 0.01,
+    ...         "momentum": 0.9,
     ...     },
     ... }
 
-    >>> parser.add_argument("--optimizer", type=Callable[[Iterable], Optimizer])  # doctest: +IGNORE_RESULT
+    >>> parser.add_argument("--optimizer", type=Callable[[Iterable, float], Optimizer])  # doctest: +IGNORE_RESULT
```

Comment:
The changes in this example have made it very misleading. The concept being introduced here is parameters that are only available after the class has been instantiated. For the case of optimizer, this is only `params`. The type here must be `Callable[[Iterable], Optimizer])`, otherwise it makes what is being explained more confusing. `SGD` can have both `lr` and `momentum`, but both of these should be given in the `init_args`.

Certainly there could be classes that have more than one parameters available after instantiation, which is what this pull request is fixing. If you want to explain that this is possible, do so after the `SGD` example. And the example shouldn't be with `Optimizer` since that is a known pytorch example that has only one such parameter.

Reply:
Could also be just a small paragraph explaining that it is possible and only give as example a type hint.

Reply:
Good point. Fixed docs by adding a small paragraph for clarification.